packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import os from 'os'
+import path from 'path'
+import archiver from 'archiver'
+import _fetch from 'node-fetch'
+import {URL} from 'url'
+import {readFile, stat, mkdtemp, rmdir} from 'fs/promises'
+import {createWriteStream} from 'fs'
+import {Minimatch} from 'minimatch'
+import git from 'git-rev-sync'
+import validator from 'validator'
+
+export const DEFAULT_CLOUD_ORIGIN = 'https://cloud.mobify.com'
+export const DEFAULT_DOCS_URL =
+    'https://developer.salesforce.com/docs/commerce/pwa-kit-managed-runtime/guide/pushing-and-deploying-bundles.html'
+
+interface Credentials {
+    username: string
+    api_key: string
+}
+
+interface CloudAPIClientOpts {
+    credentials: Credentials
+    origin?: string
+    fetch?: _fetch
+}
+
+interface StringMap {
+    [key: string]: string
+}
+
+interface Bundle {
+    message: string
+    encoding: string
+    data: string
+    ssr_parameters: object
+    ssr_only: string[]
+    ssr_shared: string[]
+}
+
+
+export const getPkgJSON = async () => {
+    const data = await readFile(path.join(__dirname, '..', '..', 'package.json'))
+    return JSON.parse(data.toString('utf-8'))
+}
+
+export class CloudAPIClient {
+    private opts: Required<CloudAPIClientOpts>
+
+    constructor(params: CloudAPIClientOpts) {
+        this.opts = {
+            origin: params.origin || DEFAULT_CLOUD_ORIGIN,
+            fetch: params.fetch || _fetch,
+            credentials: params.credentials
+        }
+    }
+
+    private getAuthHeader(): StringMap {
+        const {username, api_key} = this.opts.credentials
+        const encoded = Buffer.from(`${username}:${api_key}`, 'binary').toString('base64')
+        return {Authorization: `Basic ${encoded}`}
+    }
+
+    private async getHeaders(extras: StringMap): Promise<StringMap> {
+        const pkg = await getPkgJSON()
+        return {
+            'User-Agent': `${pkg.name}@${pkg.version}`,
+            ...this.getAuthHeader(),
+            ...extras
+        }
+    }
+
+    private async throwForStatus(res) {
```

Comment:
```suggestion
    private async throwOnHttpError(res) {
```
`throwForStatus` sounds to me like it would take two arguments, e.g. `throwForStatus(res, 404)`, and it only throws if the status is an exact match. A different name would make it more clear which statuses throw errors.